javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/AbstractJavaParserContext.java

```diff
@@ -287,4 +290,27 @@ public Optional<MethodUsage> solveMethodAsUsage(String name, List<ResolvedType>
     public N getWrappedNode() {
         return wrappedNode;
     }
+
+    public List<TypePatternExpr> typePatternExprsDiscoveredInPattern(PatternExpr patternExpr) {
```

Comment:
In the symbol solver, we're generally only interested in `TypePatternExprs` since these contain the variable declarations needed. This method just does a breadth-first traversal of the pattern tree to discover all `TypePatternExprs` contained in the nested `RecordPatternExpr` structure.